### ISSUE TITLE
 redis: ping timeout - Need custom timeout 

### ISSUE THREAD

**Issue by Dagetby** [go-redis/redis#1761]
We have a ping error falling, `redis: discarding bad PubSub connection: redis: ping timeout`. After trying to understand the problems, the team and I came to the conclusion that our service sometimes does not have time to reach redis, because we have our own custom ping and its time is every 5 seconds, and it works without errors. The problem is that, there is no function to set a custom ping time to redis, it turns out there is no way to get rid of this not significant, but annoying error.

```
const (
	pingTimeout     = time.Second
	chanSendTimeout = time.Minute
)
...
func (c *PubSub)initPing(){
...
// Need set a custom time
timer := time.NewTimer(pingTimeout)
...
}
```

Somethink like 
```
//Add timeout to struct PubSub and add method for change it 
func (c *PubSub) CustomPingTimeOut(t time.Duration){
c.pingTimeout = t
}

```

P.S. Thank you for the great libr!

**Comment by aprilito1965** [comment]
We are experiencing the same behaviour!

**Comment by DavidKmn** [comment]
Found this behaviour to be an issue too, would be great to be able to set the timeouts ourselves. Seems like a quick PR?

**Comment by monkey92t** [comment]
Yes, it looks like we should redesign pubsub.Ping.  In the case of a bad network, it caused a lot of troubles.
I will try to improve it.

**Comment by monkey92t** [comment]
Fixed by #1764
